### ISSUE TITLE
Introduce `DuplicateAnnotationListing` bug checker

### ISSUE THREAD

**Issue by mohamedsamehsalah** [PicnicSupermarket/error-prone-support#1044]
### Problem

Duplicate entries inside annotation listing are a common mistake that should be flagged.

### Description of the proposed new feature

- [x] Support a stylistic preference.
- [x] Avoid a common gotcha, or potential problem.
- [x] Improve performance.

Introduce a new _BugChecker_ to flag duplicate entries as this is a common mistake. Similar to `LexicographicalAnnotationListing`, the new _BugChecker_ should identify and handle duplicate entries to improve code quality and reduce potential errors. 

I would like to rewrite the following code:
```java
            @Foo
            @Baz
            @Baz
            A duplicateAnnotations();
```

to:
```java
            @Foo
            @Baz
            A duplicateAnnotations();
```

### Considerations
👉 **To consider:** A duplicate entry is a duplicate pointer reference and not just the same name.
👉 **Open question:** Are there situations where we do not want to flag this ?

### Participation

- [x] I am willing to submit a pull request to implement this improvement.

### Additional context

See this [comment](https://github.com/PicnicSupermarket/error-prone-support/blob/1cc792c615ff9a3fdb164506255dda7e78069ad8/error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/LexicographicalAnnotationListing.java#L41).

**Comment by Stephan202** [comment]
Thanks for filing this issue @mohamedsamehsalah! One key point of feedback: I think we're looking for an analog of [`LexicographicalAnnotationAttributeListing`](https://github.com/PicnicSupermarket/error-prone-support/blob/b39e322a671deea549aff427976fa34f8d6e929b/error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/LexicographicalAnnotationAttributeListing.java), i.e. a check that replaces `@A("foo", "bar", "foo")` with `@A("foo", "bar")`.

(Repeated annotations such as `@Baz() @Baz()` are rejected by the compiler, unless they're declared [`@Repeatable`](https://docs.oracle.com/javase/8/docs/api/java/lang/annotation/Repeatable.html), in which case the repetition likely is intended.)

**Comment by Stephan202** [comment]
W.r.t. your considerations:
* I'm not sure how pointer references factor in, but since only primitives, enum values and strings can be annotation arguments, the invariant holds that a value is a duplicated if and only if it is equal to some other value inside the same annotation attribute.
* There are likely cases we'll want to exclude, but none come to mind right now. We might want to run a first version against the Picnic code base to assess what is flagged.
